test/jit/test_enum.py

```diff
@@ -125,142 +124,181 @@ def enum_comp(x: Color, y: Color) -> bool:
             torch.jit.script(enum_const)
 
     def test_enum_ivalue_type(self):
-        global Color
+        global ColorIValueType
 
-        class Color(Enum):
+        class ColorIValueType(Enum):
             RED = 1
             GREEN = 2
 
         def is_color_enum(x: Any):
-            if isinstance(x, Color):
+            if isinstance(x, ColorIValueType):
                 return True
             else:
                 return False
 
         with torch._jit_internal._disable_emit_hooks():
             scripted_is_color_enum = torch.jit.script(is_color_enum)
 
-        self.assertEqual(scripted_is_color_enum(Color.RED), True)
-        self.assertEqual(scripted_is_color_enum(Color.GREEN), True)
+        self.assertEqual(scripted_is_color_enum(ColorIValueType.RED), True)
+        self.assertEqual(scripted_is_color_enum(ColorIValueType.GREEN), True)
         self.assertEqual(scripted_is_color_enum(1), False)
 
     def test_closed_over_enum_constant(self):
-        global Color
+        global ColorClosedOver
 
-        class Color(Enum):
+        class ColorClosedOver(Enum):
             RED = 1
             GREEN = 2
 
-        a = Color
+        a = ColorClosedOver
 
         def closed_over_aliased_type():
             return a.RED.value
 
         with torch._jit_internal._disable_emit_hooks():
             scripted = torch.jit.script(closed_over_aliased_type)
 
-        self.assertEqual(scripted(), Color.RED.value)
+        self.assertEqual(scripted(), ColorClosedOver.RED.value)
 
 
-        b = Color.RED
+        b = ColorClosedOver.RED
 
         def closed_over_aliased_value():
             return b.value
 
         with torch._jit_internal._disable_emit_hooks():
             scripted = torch.jit.script(closed_over_aliased_value)
 
-        self.assertEqual(scripted(), Color.RED.value)
+        self.assertEqual(scripted(), ColorClosedOver.RED.value)
 
     def test_enum_as_module_attribute(self):
-        global Color
+        global ColorModuleAttr
 
-        class Color(Enum):
+        class ColorModuleAttr(Enum):
             RED = 1
             GREEN = 2
 
         class TestModule(torch.nn.Module):
-            def __init__(self, e: Color):
+            def __init__(self, e: ColorModuleAttr):
                 super(TestModule, self).__init__()
                 self.e = e
 
             def forward(self):
                 return self.e.value
 
-        m = TestModule(Color.RED)
+        m = TestModule(ColorModuleAttr.RED)
+        with torch._jit_internal._disable_emit_hooks():
+            scripted = torch.jit.script(m)
+
+        self.assertEqual(scripted(), ColorModuleAttr.RED.value)
+
+    def test_enum_return(self):
+        global ColorForReturn
+
+        class ColorForReturn(Enum):
+            RED = 1
+            GREEN = 2
+
+        @torch.jit.script
+        def return_enum(cond):
+            if bool(cond):
```

Comment:
Consider using a bool here directly instead of an inferred Tensor and that is converted to a boolean later.
```suggestion
        def return_enum(cond: bool):
            if cond:
```